### ISSUE TITLE
[BUG] "select date '1993-01-01' + interval '1' year" generate wrong answer

### ISSUE THREAD

**Issue by funnyxj** [greenplum-db/gpdb#1187]
```
postgres=# select date '1993-01-01' + interval '1' year;
      ?column?
---------------------
 1993-01-01 00:00:00
(1 row)

```

The result should be '1994-01-01 00:00:00', this bug will cause TPCH Q6 get wrong result.

```
select
    sum(l_extendedprice * l_discount) as revenue
from
    lineitem
where
    l_shipdate >= date '1993-01-01'
    and l_shipdate < date '1993-01-01' + interval '1' year
    and l_discount between 0.06 - 0.01 and 0.06 + 0.01
    and l_quantity < 25
LIMIT 1;
```

**Comment by zeromax007** [comment]
I think  you can use 

> select

 date '1993-01-01' + interval **'1 year';**

**Comment by ascherbaum-pivotal** [comment]
Works as expected in PostgreSQL 9.4. It will return 1994.

**Comment by ascherbaum-pivotal** [comment]
Heikki confirmed that this is fixed in PostgreSQL 8.4. The next merge will fix this problem.

**Comment by Eulerizeit** [comment]
@melanieplageman I bet you can close this too.
